app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :author, class_name: 'User'
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
+  def five_recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 end
```

Comment:
- OPTIONAL   **Hint:**  you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html)  after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private

```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end 

```

Same can be done on likes and comment model